lxd/network/driver_bridge.go

```diff
@@ -2543,6 +2543,32 @@ func (n *bridge) getExternalSubnetInUse() ([]externalSubnetUsage, error) {
 	externalSubnets = append(externalSubnets, bridgeNetworkExternalSubnets...)
 	externalSubnets = append(externalSubnets, bridgedNICExternalRoutes...)
 
+	// Detect if there are any conflicting proxy devices on all instances with the to be created network forward
+	err = n.state.DB.Cluster.InstanceList(context.TODO(), func(inst db.InstanceArgs, p api.Project) error {
+		devices := db.ExpandInstanceDevices(inst.Devices, inst.Profiles)
+		for devName, devConfig := range devices {
+			if devConfig["type"] == "proxy" {
```

Comment:
We generally prefer using the "happy path" coding style. I.e. rather than nesting for every if condition you want, instead add checks at the top that `continue` if the item you're looking at isn't what you want. Effectively only continue if you are "happy" this is what you want.

So instead to:

```go
	for devName, devConfig := range devices {
			if devConfig["type"] != "proxy" {
                continue
           }
```

See also https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88